File: 2020/dec1.py
"""
Solution to https://adventofcode.com/2020/day/1
"""
from pathlib import Path

# from the root of the project
INPUT_FILE = Path.cwd() / "2020" / "dec1.txt"


def part1() -> int:
    lines = [int(line) for line in INPUT_FILE.read_text().split("\n") if line]
    for line1 in lines:
        for line2 in lines:
            if line1 + line2 == 2020:
                return line1 * line2

    return 0


def part2() -> int:
    lines = [int(line) for line in INPUT_FILE.read_text().split("\n") if line]
    for line1 in lines:
        for line2 in lines:
            for line3 in lines:
                if line1 + line2 + line3 == 2020:
                    return line1 * line2 * line3

    return 0

print(part1())
print(part2())

File: 2020/dec1.clj
(ns dec1
  (:require [clojure.string :refer [split-lines]]))

(def input (map #(Integer/parseInt %) (split-lines (slurp "./dec1.txt"))))

(defn part1 []
  (first (for [x input
               y input
               :when (= (+ x y) 2020)]
          (* x y))))

(defn part2 []
  (first (for [x input
               y input
               z input
               :when (= (+ x y z) 2020)]
          (* x y z))))

(= (part1) 793524)
(= (part2) 61515678)
